crates/turborepo-lockfiles/src/berry/mod.rs

```diff
@@ -266,11 +266,10 @@ impl<'a> BerryLockfile<'a> {
                     resolutions.insert(dependency, dep_locator.clone());
                 }
 
-                if let Some(descriptors) = reverse_lookup.get(locator) {
-                    for descriptor in descriptors {
-                        resolutions.insert((*descriptor).clone(), locator.clone());
-                    }
-                }
+                // Included workspaces will always have their locator listed as a descriptor.
+                // All other descriptors should show up in the other workspace package
+                // dependencies.
+                resolutions.insert(Descriptor::from(locator.clone()), locator.clone());
```

Comment:
```
"c@*, c@workspace:*, c@workspace:packages/c":
```

Am I understanding correctly that this whole line is the "descriptor" and the "locator" is the `packages/c` part?

Reply:
Kinda: `c@*`, `c@workspace:*`, and `c@workspace:packages/c` are all descriptors and then they all get concat'd to be a key in the YAML lockfile. `c@workspace:packages/c` is *also* a locator in this lockfile which appears in the `resolution` field of the lockfile entry.